packages/edit-site/src/components/page-templates/template-actions.js

```diff
@@ -87,31 +110,83 @@ export const deleteTemplateAction = {
 	id: 'delete-template',
 	label: __( 'Delete template' ),
 	isPrimary: true,
+	isBulk: true,
 	icon: trash,
 	isEligible: isTemplateRemovable,
 	hideModalHeader: true,
-	RenderModal: ( { item: template, closeModal } ) => {
+	RenderModal: ( { items: templates, closeModal } ) => {
 		const { removeTemplate } = useDispatch( editSiteStore );
+		const { createSuccessNotice, createErrorNotice } =
+			useDispatch( noticesStore );
+		const { deleteEntityRecord } = useDispatch( coreStore );
 		return (
 			<VStack spacing="5">
 				<Text>
-					{ sprintf(
-						// translators: %s: The template or template part's title.
-						__( 'Are you sure you want to delete "%s"?' ),
-						decodeEntities( template.title.rendered )
-					) }
+					{ templates.length > 1
+						? sprintf(
+								// translators: %s: The template or template part's title.
```

Comment:
This comment should refer to the number of items: `translators: The number of items.`